src/neo/Wallets/Wallet.cs

```diff
@@ -298,6 +301,32 @@ public Transaction MakeTransaction(TransferOutput[] outputs, UInt160 from = null
                                 }
                                 sb.EmitAppCall(output.AssetId, "transfer", account, output.ScriptHash, value);
                                 sb.Emit(OpCode.ASSERT);
+
+                                // Solid transfer
+
+                                if (output.SolidTransfer)
+                                {
+                                    byte[] solidTransferScript;
+                                    using (ScriptBuilder sbSolid = new ScriptBuilder())
+                                    {
+                                        sbSolid.EmitPush(value);
+                                        sbSolid.EmitAppCall(output.AssetId, "balanceOf", account);
```

Comment:
If so, we also need to update here, `CallFlags.AllowStates` -> `CallFlags.ReadOnly`.
https://github.com/neo-project/neo/blob/22f80a7ccfb00af15006569921ed87536cf3f857/src/neo/SmartContract/Helper.cs#L177-L179

Reply:
IMO it's much easier to unconditionally do `callFlags = CallFlags.ReadOnly` and allow any verification scripts to access storage. This would then allow this feature to be a wallet feature not requiring anything special support from the ledger and allowing to remove `IsSolidTransfer` helper completely (except fee calculation?). It would also of course allow to have other interesting state-dependent verification scripts. 

Witness StateDependent flag could then be set based on `IsStandardContract()`, keeping this optimization in place for the most common case.

Reply:
Completely agree

Reply:
Also agree... from the perspective on Neo, there are so many applications that require reading from storage, and it's not good to make exceptions to each of them. Best approach is to allow storage reads, and deal with the fact that transaction ordering will affect future verifications.